rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/misc.kt

```diff
@@ -223,6 +223,36 @@ internal fun FileDefinition.toDataDefinitions(): List<DataDefinition> {
     return dataDefinitions
 }
 
+private fun FileDefinition.loadDisplayFileMetadata(): List<DataDefinition> {
+    val dataDefinitions = mutableListOf<DataDefinition>()
+    val dspfConfig = MainExecutionContext.getConfiguration()
+        .dspfConfig ?: error("Not found metadata for $this because missing property dspfConfig in configuration")
+    val metadata = kotlin.runCatching {
+        dspfConfig.metadataProducer.invoke(name)
+    }.onFailure { error ->
+        error("Not found display file for $this", error)
+    }.getOrNull() ?: error("Not found display file for $this")
+
+    // in a file declaration of type WORKSTN the executed record name is never available
+    // internalFormatName is always null
+    // record name will be available further in EXFMT statement
+
+    dataDefinitions.addAll(
+        metadata.fields.map { dspfField ->
+            dspfField.toDataDefinition(prefix = prefix, position = position).apply {
+                createDbFieldDataDefinitionRelation(dspfField.fieldName, name)
+            }
+        }
+    )
+
+    return dataDefinitions
+}
+
+internal fun FileDefinition.toDataDefinitions(): List<DataDefinition> {
+    if (fileType == FileType.DB) return loadDbFileMetadata()
```

Comment:
Use `when` and throw an error if `fileType` is neither `FileType.DB` nor `FileType.VIDEO`.